src/main/java/wolox/training/models/User.java

```diff
@@ -62,12 +63,8 @@ public void setBirthdate(LocalDate birthdate) {
         this.birthdate = birthdate;
     }
 
-    public List<Book> getBooks() {
-        return (List<Book>) Collections.unmodifiableList(books);
-    }
-
-    public void setBooks(List<Book> books) {
-        this.books = books;
+    public List<Optional<Book>> getBooks() {
```

Comment:
Remember to return a list of books rather than a list of optionals 😅  (if the book is not present, it won't be added to the list)